tests/integration/snapshot/v3_snapshot_test.go

```diff
@@ -286,10 +286,23 @@ func restoreCluster(t *testing.T, clusterN int, dbPath string) (
 // TODO: TLS
 func newEmbedURLs(n int) (urls []url.URL) {
 	urls = make([]url.URL, n)
+	allocatedPorts := make(map[int]struct{})
 	for i := 0; i < n; i++ {
-		rand.Seed(int64(time.Now().Nanosecond()))
-		u, _ := url.Parse(fmt.Sprintf("unix://localhost:%d", rand.Intn(45000)))
+		u, _ := url.Parse(fmt.Sprintf("unix://localhost:%d", allocateUniquePort(2000, 45000, allocatedPorts)))
 		urls[i] = *u
 	}
 	return urls
 }
+
+func allocateUniquePort(minPort, maxPort int, allocatedPort map[int]struct{}) int {
```

Comment:
we're not guaranteed to get a freeport this way - and it's more state to track, which won't scale too well when expanding across test cases

we might wanna do something more like:

```
t.Helper()
addr, err := net.ResolveTCPAddr("tcp", "localhost:0")
if err != nil {
	t.Failf("failed to resolve localhost: %v", err)
}

l, err := net.ListenTCP("tcp", addr)
defer l.Close()
if err != nil {
	t.Failf("failed to listen tcp: %v", err)
}
return l.Addr().(*net.TCPAddr).Port
```

Reply:
It isn't worse than current implementation. Note that this is only used for snapshot related cases. Let's unify the port allocation in a separate PR.

Reply:
@ptabor  already proposed a similar solution above.

Reply:
It's not worse - but it's not a marked improvement, we could use this to prove out a better implementation that can then migrate to be shared everywhere.

State tracking when a ~simpler solution would be more reliable doesn't really seem too necessary?

Reply:
Done.